From 85db70336ab66136481926bcd7f5abe599e2aa4f Mon Sep 17 00:00:00 2001
From: Richard Zhu <Richard.Zhu@freescale.com>
Date: Mon, 2 Nov 2015 11:41:02 +0800
Subject: [PATCH] MLK-11803 pci: imx: imx6qp pcie ep self-test failed

imx pcie used the wrab mode to do the cached access
methods on axi bus. There is 64bytes address mis-aligned
problem.
Disable the cached operations.

Signed-off-by: Richard Zhu <Richard.Zhu@freescale.com>
---
 drivers/pci/host/pci-imx6.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/pci/host/pci-imx6.c b/drivers/pci/host/pci-imx6.c
index 90c653d..405c184 100644
--- a/drivers/pci/host/pci-imx6.c
+++ b/drivers/pci/host/pci-imx6.c
@@ -1412,11 +1412,7 @@ static int __init imx6_pcie_probe(struct platform_device *pdev)
 		 * reserved ddr memory after write the ddr_test_region
 		 * content to rc.
 		 */
-		if (is_imx7d_pcie(imx6_pcie))
-			pcie_arb_base_addr = ioremap_nocache(pp->mem_base,
-					test_region_size);
-		else
-			pcie_arb_base_addr = ioremap_cache(pp->mem_base,
+		pcie_arb_base_addr = ioremap_nocache(pp->mem_base,
 					test_region_size);
 
 		if (!pcie_arb_base_addr) {
-- 
1.8.0

